### PR TITLE
[Chore] Create example Modin RayJob

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Why are these changes needed?

This PR creates an example RayJob for Modin tasks.
Related PR: https://github.com/ray-project/ray/pull/46345

## Related issue number

Resolves: #2190

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [ ] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/config/samples/ray-job.modin.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    import urllib.request
+    import ray
+    import numpy as np
+    import modin.pandas as pd
+    import modin.config as modin_cfg
+    from modin.config import BenchmarkMode
+
+    ray.init()
+
+    @ray.remote
```
[the remark this answers is not in the document]
Comment (MortalHappiness) [reply ray-operator/config/samples/ray-job.modin.yaml]: Unwraped

File: ray-operator/config/samples/ray-job.modin.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        minReplicas: 1
+        maxReplicas: 5
+        groupName: small-group
+        rayStartParams: {}
+        template:
+          spec:
+            containers:
+              - name: ray-worker
+                image: rayproject/ray:2.31.0
+                lifecycle:
```
[the remark this answers is not in the document]
Comment (MortalHappiness) [reply ray-operator/config/samples/ray-job.modin.yaml]: Done

File: ray-operator/config/samples/ray-job.modin.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  name: rayjob-sample
+spec:
+  entrypoint: python /home/ray/samples/sample_code.py
+  runtimeEnvYAML: |
+    pip:
+      - modin[all]==0.31.0
+  rayClusterSpec:
+    rayVersion: "2.31.0"
+    headGroupSpec:
+      rayStartParams:
```
[the remark this answers is not in the document]
Comment (MortalHappiness) [reply ray-operator/config/samples/ray-job.modin.yaml]: Done

File: ray-operator/config/samples/ray-job.modin.yaml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                  name: code-sample
+          volumes:
+            - name: code-sample
+              configMap:
+                name: ray-job-code-sample
+                items:
+                  - key: sample_code.py
+                    path: sample_code.py
+    workerGroupSpecs:
+      - replicas: 1
```
Comment (kevin85421) [inline ray-operator/config/samples/ray-job.modin.yaml:42]: does this script run anything on Ray worker?

File: ray-operator/config/samples/ray-job.modin.yaml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    import modin.config as modin_cfg
+    from modin.config import BenchmarkMode
+
+    ray.init()
+
+    print("Modin Engine:", modin_cfg.Engine.get())
+    BenchmarkMode.put(True)
+
+    url_path = "https://modin-datasets.intel.com/green-taxi/green_tripdata_2015-01.csv"
+    df = pd.read_csv(url_path, parse_dates=["tpep_pickup_datetime", "tpep_dropoff_datetime"], quoting=3)
```
Comment (kevin85421) [inline ray-operator/config/samples/ray-job.modin.yaml:78]: will this function download the CSV on both the head and worker?

File: ray-operator/config/samples/ray-job.modin.yaml (hunk 4 above)
Comment (MortalHappiness) [reply ray-operator/config/samples/ray-job.modin.yaml:42]: No. Maybe I can delete the workerGroupSpecs

File: ray-operator/config/samples/ray-job.modin.yaml (hunk 5 above)
Comment (MortalHappiness) [reply ray-operator/config/samples/ray-job.modin.yaml:78]: No. There is nothing run on the worker as mentioned above. I'll delete the workerGroupsSpecs.

File: ray-operator/config/samples/ray-job.modin.yaml (hunk 4 above)
Comment (MortalHappiness) [reply ray-operator/config/samples/ray-job.modin.yaml]: @kevin85421 I've deleted the workerGroupSpecs. Thanks.

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**